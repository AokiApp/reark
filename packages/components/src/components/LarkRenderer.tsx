import React, { useContext } from "react";
import { LarkApiContext } from "../contexts/LarkApiContext";
import { BlockStoreProvider } from "../contexts/BlockStoreContext";
import { BlockComponent } from "./BlockComponent";

export const LarkRenderer: React.FC = () => {
  const { blocks } = useContext(LarkApiContext);

  if (!blocks || blocks.length === 0) return null;

  const rootBlock =
    blocks.find((block) => block.block_type === 1 && !block.parent_id) ||
    blocks[0];

  if (!rootBlock) return null;

  return (
    <BlockStoreProvider items={blocks}>
      <BlockComponent blockId={rootBlock.block_id} />
    </BlockStoreProvider>
  );
};
