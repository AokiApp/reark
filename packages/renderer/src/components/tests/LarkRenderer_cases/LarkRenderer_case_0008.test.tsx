import { render } from "@testing-library/react";
import { LarkRenderer } from "../../LarkRenderer";

describe("LarkRenderer 属性バリエーション: 組み合わせ違い", () => {
  it("textブロック: 太字＋下線、斜体＋打消し、インラインコード＋コメント付き", () => {
    const block = {
      block_id: "attr-combo",
      block_type: 2,
      parent_id: "",
      text: {
        elements: [
          {
            text_run: {
              content: "太字＋下線",
              text_element_style: { bold: true, underline: true },
            },
          },
          {
            text_run: {
              content: "斜体＋打消し",
              text_element_style: { italic: true, strikethrough: true },
            },
          },
          {
            text_run: {
              content: "インラインコード＋コメント付き",
              text_element_style: { inline_code: true, comment_ids: ["c2"] },
            },
          },
        ],
        style: { align: "left", folded: false },
      },
      children: [],
    };
    const blocks = [block];
    const { container } = render(<LarkRenderer initialData={{ blocks }} />);
    expect(container).toMatchSnapshot();
  });
});
