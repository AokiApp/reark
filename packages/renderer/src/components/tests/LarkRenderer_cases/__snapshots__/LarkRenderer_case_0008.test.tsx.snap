// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LarkRenderer 属性バリエーション: 組み合わせ違い textブロック: 太字＋下線、斜体＋打消し、インラインコード＋コメント付き 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-text-block"
      style="text-align: left;"
    >
      <span
        class="reark-text--bold reark-text--underline reark-text"
      >
        太字＋下線
      </span>
      <span
        class="reark-text--italic reark-text--strikethrough reark-text"
      >
        斜体＋打消し
      </span>
      <div
        class="reark-comment"
        data-comment-ids="c2"
      >
        <span
          class="reark-text--inline-code reark-text"
        >
          インラインコード＋コメント付き
        </span>
      </div>
    </div>
  </div>
</div>
`;
